submissions/src/main/kotlin/gov/cdc/prime/reportstream/submissions/controllers/SubmissionController.kt

```diff
@@ -0,0 +1,228 @@
+) {
+    /**
+     * Submits a report.
+     *
+     * This endpoint accepts reports in HL7 V2 and FHIR NDJSON formats. The report data is uploaded
+     * to Azure Blob Storage, queued for further processing, and its metadata is stored in Azure Table Storage.
+     * A custom event is also tracked in Application Insights for monitoring.
+     *
+     * @param headers the HTTP headers of the request
+     * @param contentType the content type of the report (must be "application/hl7-v2" or "application/fhir+ndjson")
+     * @param clientId the ID of the client submitting the report. Should represent org.senderName
+     * @param data the report data
+     * @return a ResponseEntity containing the reportID, status, and timestamp
+     */
+    @PostMapping("/api/v1/reports", consumes = ["application/hl7-v2", "application/fhir+ndjson"])
+    fun submitReport(
+        @RequestHeader headers: Map<String, String>,
+        @RequestHeader("Content-Type") contentType: String,
+        @RequestHeader("client_id") clientId: String,
+        @RequestHeader("content-length") contentLength: String,
+        @RequestHeader("x-azure-clientip") senderIp: String,
+        @RequestHeader(value = "payloadName", required = false) payloadName: String?,
+        @RequestBody data: String,
+    ): ResponseEntity<*> {
+        val reportId = UUID.randomUUID()
+        val reportReceivedTime = OffsetDateTime.now()
+        val contentTypeMime = contentType.substringBefore(';')
+        val status = "Received"
+        logger.info("Received report submission: reportId=$reportId, contentType=$contentTypeMime" +
+            ", clientId=$clientId${payloadName?.let { ", payloadName=$it" } ?: ""}}")
+
+        // Convert data to ByteArray
+        val dataByteArray = data.toByteArray()
+        logger.debug("Converted report data to ByteArray")
+
+        // Upload to blob storage
+        val blobClient = blobContainerClient.getBlobClient(formBlobName(reportId, contentTypeMime, clientId))
+        blobClient.upload(dataByteArray.inputStream(), dataByteArray.size.toLong())
+        logger.info("Uploaded report to blob storage: blobUrl=${blobClient.blobUrl}")
+
+        // Create the message for the queue
+        val message = SubmissionQueueMessage(reportId, blobClient.blobUrl, filterHeaders(headers))
+        val objectMapper = jacksonObjectMapper().registerModule(JavaTimeModule())
+        val messageString = objectMapper.writeValueAsString(message)
+        logger.debug("Created message for queue")
+
+        // Upload to Queue
+        queueClient.createIfNotExists()
+        queueClient.sendMessage(messageString)
+        logger.info("Sent message to queue: queueName=${queueClient.queueName}")
+
+        // Insert into Table
+        // TableEntity() sets PartitionKey and RowKey. Both are required by azure and combine to create the PK
+        val tableEntity = TableEntity(reportReceivedTime.toString(), reportId.toString())
```

Comment:
I would maybe switch this to using `Instant` rather than `Offset`.  I think our containers will always run in the east coast data center but if that changed I think the partition would be less effective with different offsets